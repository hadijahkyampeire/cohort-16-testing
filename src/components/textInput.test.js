import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, configure } from 'enzyme';

import TextInput from './textInput';

configure({ adapter: new Adapter() });

describe('Add input component', () => {
  const textInput = shallow(<TextInput />);
  it('renders properly', () => {
    expect(textInput).toMatchSnapshot();
  });

  it('handles input changes and changes state', () => {
    textInput.find('input').simulate('change', { target: { name: 'text', value: 'test' } });
    const state = textInput.state();
    expect(state.text).toEqual('test');
  });
  it('displays the text in state when the display button is called', () => {
    textInput.setState({ text: 'new text' });
    textInput.find('.btn-primary').simulate('click');
    const state = textInput.state();
    expect(state.result).toEqual('new text');
  });
});
