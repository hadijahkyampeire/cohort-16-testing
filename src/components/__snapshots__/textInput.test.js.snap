// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add input component renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          Some text:
        </span>,
        <input
          aria-describedby="emailHelp"
          className="form-control"
          id="text"
          name="text"
          onChange={[Function]}
          placeholder="Enter text"
          type="text"
        />,
        <button
          className="btn btn-primary"
          onClick={[Function]}
        >
          Display
        </button>,
        <span>
          <u>
            
          </u>
        </span>,
      ],
      "className": "text",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Some text:",
        },
        "ref": null,
        "rendered": "Some text:",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-describedby": "emailHelp",
          "className": "form-control",
          "id": "text",
          "name": "text",
          "onChange": [Function],
          "placeholder": "Enter text",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Display",
          "className": "btn btn-primary",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Display",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <u>
            
          </u>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "u",
        },
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            Some text:
          </span>,
          <input
            aria-describedby="emailHelp"
            className="form-control"
            id="text"
            name="text"
            onChange={[Function]}
            placeholder="Enter text"
            type="text"
          />,
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            Display
          </button>,
          <span>
            <u>
              
            </u>
          </span>,
        ],
        "className": "text",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Some text:",
          },
          "ref": null,
          "rendered": "Some text:",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-describedby": "emailHelp",
            "className": "form-control",
            "id": "text",
            "name": "text",
            "onChange": [Function],
            "placeholder": "Enter text",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Display",
            "className": "btn btn-primary",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Display",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <u>
              
            </u>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "u",
          },
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
